src/docfx/build/page/BuildPage.cs

```diff
@@ -16,33 +16,24 @@ public static async Task<List<Error>> Build(Context context, Document file)
         {
             Debug.Assert(file.ContentType == ContentType.Page);
 
-            var (errors, pageModel, inputMetadata) = await Load(context, file);
-            var (generateErrors, outputMetadata) = await GenerateOutputMetadata(context, file, inputMetadata);
-            errors.AddRange(generateErrors);
+            var (errors, pageModel) = await Load(context, file);
+            var (monikerError, monikers) = context.MonikerProvider.GetFileLevelMonikers(file);
+            errors.AddIfNotNull(monikerError);
 
-            var outputPath = file.GetOutputPath(outputMetadata.Monikers, file.Docset.SiteBasePath, file.IsPage);
+            var outputPath = file.GetOutputPath(monikers, file.Docset.SiteBasePath, file.IsPage);
 
             object output = null;
```

Comment:
consider a function over if else:
```suggestion
            var (outputErrors, output, metadata) = file.IsPage ? await CreatePageOutput() : CreateDataOutput();
```